src/pages/admin/tokens/edit.tsx

```diff
@@ -385,6 +387,16 @@ const AdminEditTokens = ({ marketplace }: AdminEditTokensProps) => {
                             if (e.key !== 'Enter') {
                               return
                             }
+                            if (!tokenMap.has(value)) {
+                              toast.error('Token not found')
+                              return
+                            }
+                            if (
+                              fields.some((token) => token.address === value)
+                            ) {
+                              toast.error('Token already added')
+                              return
+                            }
```

Comment:
resolver is where validations should go in the future but this certainly works for this case.

https://react-hook-form.com/ts#Resolver